src-built-in/components/toolbar/stores/toolbarStore.js

```diff
@@ -225,6 +225,23 @@ class _ToolbarStore {
 		FSBL.Clients.HotkeyClient.addGlobalHotkey(["ctrl", "alt", "h"], () => {
 			self.hideToolbar();
 		});
+
+		fin.desktop.Window.getCurrent().addEventListener("close-requested", () => this.closeRequestedHandler());
+	}
+
+	/**
+	 * handles a close request by showing the user a dialog prompting for shutdown.
+	 * affirmative will shutdown finsemble, negative will do nothing.
+	 */
+	async closeRequestedHandler() {
+		const args = await this.confirmCloseToolbar();
+		if (args.err) {
+			Logger.system.log(`Error received on confirm close: ${args.err} Continuing.`);
+		}
+		if (args.result.choice === 'affirmative') {
```

Comment:
Should it really continue and get to this line if there is `args.err`?

Reply:
I figure if there is both an error, and an affirmative shutdown, then the affirmative shutdown is important enough to just proceed despite the error.

Reply:
Okay I was thinking that maybe if there is an error then `args.result` will not be there.

Reply:
It's definitely possible, but in that case it would just log the error and do nothing.